### PR TITLE
Add Reissue of assets

### PR DESCRIPTION
**Pull request by blondfrogs**
This completes part of the Phase 2 requirements that allow the ability to reissue assets.

### CODE REVIEW

File: src/assets/assets.cpp [hunk 1]
```diff
@@ -268,6 +268,18 @@ bool AssetFromTransaction(const CTransaction& tx, CNewAsset& asset, std::string&
     return AssetFromScript(scriptPubKey, asset, strAddress);
 }
 
+bool ReissueAssetFromTransaction(const CTransaction& tx, CReissueAsset& reissue, std::string& strAddress)
+{
+    // Check to see if the transaction is an new asset issue tx
```
Comment (cfox) [inline src/assets/assets.cpp]: misleading comment

File: src/assets/assets.cpp [hunk 2]
```diff
@@ -382,16 +394,41 @@ bool OwnerAssetFromScript(const CScript& scriptPubKey, std::string& assetName, s
...
+        return false;
+    }
+
+    return true;
+}
+
 bool CTransaction::IsNewAsset() const
 {
-    // Issuing a new Asset must contain at least 3 CTxOut ( 500 Raven Burn, Any Number of other Outputs ..., Issue Asset Tx, Asset Metadata)
+    // Issuing a new Asset must contain at least 3 CTxOut ( Raven Burn Tx, Any Number of other Outputs ..., Issue Asset Tx, Owner Asset Tx)
```
Comment (cfox) [inline src/assets/assets.cpp]: misleading comment

File: src/assets/assets.cpp [hunk 3]
```diff
@@ -742,6 +892,116 @@ bool CAssetsCache::AddNewAsset(const CNewAsset& asset, const std::string address
...
+
+    // If the undid amount is now 0. Remove the address from the set of addresses
+    if (mapAssetsAddresses.count(reissue.strName) && mapAssetsAddressAmount[pair] == 0) {
+        mapAssetsAddresses.at(reissue.strName).erase(address);
+    }
+
+    // Change the asset data by undoing what was reissued
+    assetData.nAmount -= reissue.nAmount;
+    assetData.nReissuable = 1;
+    // TODO get the old ipfs hash from the undoblock
```
Comment (cfox) [inline src/assets/assets.cpp]: is this a TODONE?

File: src/assets/assets.cpp [hunk 4]
```diff
@@ -1048,6 +1358,28 @@ bool CheckIssueBurnTx(const CTxOut& txOut)
     return true;
 }
 
+bool CheckReissueBurnTx(const CTxOut& txOut)
+{
+    // Check the first transaction is the 500 Burn amount to the burn address
```
Comment (cfox) [inline src/assets/assets.cpp]: 500 for reissue or 100?

File: src/rpc/assets.cpp [hunk 5]
```diff
@@ -487,6 +487,154 @@ UniValue transfer(const JSONRPCRequest& request)
...
+                "2. \"address\"                  (string, required) address to send the asset to\n"
+                "3. \"amount\"                   (number, required) number of assets to reissue\n"
+                "4. \"reissuable\"               (boolean, optional, default=true), whether future reissuance is allowed\n"
+                "5. \"new_ifps\"                  (string, optional, default= not changing), whether to update the current ipfshash\n"
+
+                "\nResult:\n"
+                "\"txid\"                     (string) The transaction id\n"
+
+                "\nExamples:\n"
+                + HelpExampleCli("transfer", "\"asset_name\" \"address\" \"20\"")
```
Comment (cfox) [inline src/rpc/assets.cpp]: these two example are copypasta

**Comment by blondfrogs** [comment]
Updated the pull request so it doesn't contain misleading comments.

**Pull request merged**